ray-operator/Dockerfile.buildx

```diff
@@ -1,4 +1,4 @@
-FROM registry.access.redhat.com/ubi9/ubi-minimal:9.3
+FROM registry.access.redhat.com/ubi9/ubi-minimal:9.4
```

Comment:
@kevin85421 is there a particular reason we're still using this base image?

Reply:
No reason - we just forget to update it in https://github.com/ray-project/kuberay/pull/1967. Maybe we can use `golang:1.22.4-bullseye` instead? cc @thomasdesr